### PR TITLE
EditorVR internal re-org (pass 2)

### PR DESCRIPTION
**Pull request by amirebrahimi**

### CODE REVIEW

File: Scripts/Core/EditorVR.Viewer.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#if UNITY_EDITORVR
```
Comment (mtschoen-unity) [inline Scripts/Core/EditorVR.Viewer.cs]: I think we should call the file/class PlayArea, not Viewer. ViewerPivot sort of makes sense, but I think Play Area is more explicit about what you use it for.

File: Scripts/Input/OVRTouchInputToEvents.cs [hunk 2]
```diff
@@ -1,164 +1,183 @@
...
-	public bool active { get; private set; }
-
-#if ENABLE_OVR_INPUT
-	public void Update()
-	{
-		// Manually update the Touch input
-		OVRInput.Update();
-
-		if ((OVRInput.GetActiveController() & OVRInput.Controller.Touch) == 0)
+		public bool active
```
Comment (mtschoen-unity) [inline Scripts/Input/OVRTouchInputToEvents.cs]: Can this go in a base class?

File: Scripts/Core/EditorVR.Viewer.cs (hunk 1 above)
Comment (amirebrahimi) [inline Scripts/Core/EditorVR.Viewer.cs]: Rename references to ViewerPivot to CameraRig

File: Scripts/Core/EditorVR.DirectSelection.cs [hunk 3]
```diff
@@ -1,221 +1,152 @@
...
-			cameraDiff.y = 0;
-			var rotationOffset = rotationDiff * cameraDiff - cameraDiff;
-
-			var endPosition = viewerPivot.position + (playerHead.position - mainCamera.position) + rotationOffset;
-			var endRotation = viewerPivot.rotation * rotationDiff;
-			var startTime = Time.realtimeSinceStartup;
-			var diffTime = 0f;
-
-			while (diffTime < kViewerPivotTransitionTime)
+			internal void OnObjectGrabbed(GameObject selection)
```
Comment (amirebrahimi) [inline Scripts/Core/EditorVR.DirectSelection.cs]: VS oddity, but technically would compile with static.

File: Scripts/Core/EditorVR.cs [hunk 4]
```diff
@@ -1,7 +1,6 @@
 #if UNITY_EDITOR
 //#define ENABLE_MINIWORLD_RAY_SELECTION
```
Comment (amirebrahimi) [inline Scripts/Core/EditorVR.cs]: Let's remove this.

File: Scripts/Core/EditorVR.cs [hunk 5]
```diff
@@ -38,23 +37,83 @@ partial class EditorVR
...
+			public ITool currentTool;
+			public IMenu customMenu;
+			public PinnedToolButton previousToolButton;
+			public readonly Dictionary<IMenu, Menus.MenuHideFlags> menuHideFlags = new Dictionary<IMenu, Menus.MenuHideFlags>();
+			public readonly Dictionary<IMenu, float> menuSizes = new Dictionary<IMenu, float>();
+		}
+
+		class Nested
+		{
+			public static EditorVR evr { protected get; set; }
```
Comment (amirebrahimi) [inline Scripts/Core/EditorVR.cs]: Constrained to EditorVR, so easy to update if we need that later.

File: Scripts/Modules/ObjectModule.cs [hunk 6]
```diff
@@ -1,22 +1,30 @@
 ﻿using System;
```
Comment (amirebrahimi) [inline Scripts/Modules/ObjectModule.cs]: Also rename PlaceObject

File: Scripts/Modules/MultipleRayInputModule/MultipleRayInputModule.cs [hunk 7]
```diff
@@ -95,7 +95,7 @@ public override void Process()
 			// We don't process with all other input modules because we need fine-grained control to consume input
 		}
 
-		public void ProcessInput(ActionMapInput input, Action<InputControl> consumeControl)
+		public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
```
Comment (amirebrahimi) [inline Scripts/Modules/MultipleRayInputModule/MultipleRayInputModule.cs]: Moving towards a dictionary of IProcessInputs, so we'll keep this.

**Pull request merged**